cmd/tk/main.go

```diff
@@ -54,10 +55,37 @@ func main() {
 		prefixCommands("tk-")...,
 	)
 
+	outputPprofFile := os.Getenv("TK_PPROF_FILE")
+
+	// So that the other defers can run
+	exitCode := 0
+	defer func() {
+		os.Exit(exitCode)
+	}()
+
+	if outputPprofFile != "" {
+		pprofFile, err := os.Create(outputPprofFile)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error creating pprof file: %s\n", err)
+			exitCode = 1
+			return
+		}
+		defer pprofFile.Close()
+
+		err = pprof.StartCPUProfile(pprofFile)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error starting pprof: %s\n", err)
```

Comment:
This error message could also be set as a variable and printed in the exit defer. It's always the same pattern. Could even get rid of the exitCode var and `exit 1` if the error is not nil